instrumentation/google_apis_core/lib/opentelemetry/instrumentation/google_apis_core/patches/http_command.rb

```diff
@@ -0,0 +1,54 @@
+# frozen_string_literal: true
+
+# Copyright The OpenTelemetry Authors
+#
+# SPDX-License-Identifier: Apache-2.0
+
+module OpenTelemetry
+  module Instrumentation
+    module GoogleApisCore
+      module Patches
+        # Module to prepend to ::Google::Apis::Core::HttpCommand for instrumentation
+        module HttpCommand
+          def opencensus_begin_span
+            return if @opentelemetry_tracing_span
+            return unless OpenTelemetry::Trace.current_span.context.trace_flags.sampled?
```

Comment:
I know we did this in the Shopify instrumentation, to match the check in the method we're monkey-patching, but I don't think that's really in the spirit of OpenTelemetry. If we're trying to avoid overhead from instrumentation when we're not recording, the supported way to do that is:
```suggestion
            return unless OpenTelemetry::Trace.current_span.recording?
```

@dazuma do you have an opinion on this?